File: Java/Language/Generics/TypeParamTest2/Program1.java
class Program {

	private static void printStringStack(SimpleStack<String> store) {
		while(!store.empty())
			System.out.println(store.pop());
		System.out.println("----------------------");
	}

	private static void printIntervalStack(SimpleStack<Interval> store) {
		while(!store.empty())
			System.out.println(store.pop());
		System.out.println("----------------------");
	}

	public static void main(String[] args) {
		SimpleStack<String> a = new SimpleStack<String>();
		a.push("Monday");
		a.push("Tuesday");
		a.push("Wednesday");
		a.push("Thursday");
		a.push("Friday");
		//a.push(123.45);
		printStringStack(a);
		SimpleStack<Interval> b = new SimpleStack<Interval>();
		b.push(new Interval(5, 31));
		b.push(new Interval(7, 42));
		b.push(new Interval(6, 13));
		b.push(new Interval(4, 24));
		printIntervalStack(b);
	}
}

File: dotnet/Language/Generics/TypeParamTest2/DemoApp/Program1.cs
class Program
{
    private static void PrintStack(SimpleStack<string> store)
    {
        while(!store.Empty())
            Console.WriteLine(store.Pop());
    }

    private static void PrintStack(SimpleStack<Interval> store)
    {
        while(!store.Empty())
            Console.WriteLine(store.Pop());
    }

    public static void Main()
    {
        SimpleStack<string> a = new SimpleStack<string>();
        a.Push("Monday");
        a.Push("Tuesday");
        a.Push("Wednesday");
        a.Push("Thursday");
        a.Push("Friday");
        //a.Push(123.4);
        PrintStack(a);
        Console.WriteLine("---------------");
        SimpleStack<Interval> b = new();
        b.Push(new Interval(5, 31));
        b.Push(new Interval(7, 42));
        b.Push(new Interval(6, 53));
        b.Push(new Interval(4, 14));
        PrintStack(b);
    }
}